wpilibNewCommands/src/test/native/cpp/frc2/command/AddRequirementsTest.cpp

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#include <wpi/array.h>
+
+#include "CommandTestBase.h"
+#include "frc2/command/Command.h"
+#include "frc2/command/RunCommand.h"
+
+using namespace frc2;
+
+class OverloadResolver {
+ public:
+  OverloadResolver(std::function<void()> requirementsOverload,
+                   std::function<void()> smallSetOverload,
+                   std::function<void()> subsystemOverload)
+      : m_requirementsOverload{std::move(requirementsOverload)},
+        m_smallSetOverload{std::move(smallSetOverload)},
+        m_subsystemOverload{std::move(subsystemOverload)} {}
+
+  void AddRequirements(Requirements requirements) { m_requirementsOverload(); }
+
+  void AddRequirements(wpi::SmallSet<Subsystem*, 4> requirements) {
+    m_smallSetOverload();
+  }
+
+  void AddRequirements(Subsystem* requirement) { m_subsystemOverload(); }
```

Comment:
Is it possible to do this by mocking the actual `Command` class (or using it in some other way) rather than duplicating it to a completely separate class?

Reply:
I looked into differentiating between overloads when mocking and it didn't seem possible, but I might find something later.

Reply:
Unfortunately, `AddRequirements` isn't marked virtual in `Command`, which really limits the ability to mock it. In fact, `MockAddRequirementsCommand` really gets nothing out of subclassing `Command`, since it can only hide `Command::AddRequirements`, not override it (so e.g. if it gets passed to a method taking Command*, that method would only call `Command`'s implementations). I definitely like the mocking option over the callbacks, but I'm not sure this really should derive from `Command`.

Reply:
Do what you think should be done.